src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -347,6 +363,22 @@ void VfsCfApi::onHydrationJobFinished(HydrationJob *job)
     }
 }
 
+void VfsCfApi::onHydrationJobCanceled(HydrationJob *job)
+{
+    const auto folderPath = job->localPath();
+    const auto folderRelativePath = job->folderPath();
+
+    // Remove placeholder file because there might be already pumped
+    // some data into it
+    QFile::remove(folderPath + folderRelativePath);
```

Comment:
This is surprising. Isn't the data committed on disk by cfapi only when the request completes? I'd be surprised they'd go through all the trouble to prevent the sync engine from writing directly in the file itself to then leave inconsistent data behind...

Reply:
If you cancel a request and then look at the file size you can see that the file size is not zero bytes anymore. But I'm not sure if there is really already data pumped in. I thought it would be cleaner to just remove it and replace it with an empty placeholder.

Reply:
> If you cancel a request and then look at the file size you can see that the file size is not zero bytes anymore.

I think you're slightly confused here. It's already non-zero *before* the request. The placeholders expose the size of the file as reported by the server metadata even if the file isn't hydrated.

Please test again without that removal and db + placeholder mangling. I mean when the client *has to* do this kind of things sure, but if that's not required then it's better not too, it's always a risk of ending up with inconsistent data if something fails during the course of such actions.

Reply:
@er-vin Actually, I have noticed what @FlexW mentioned. There is this field called "Size on disk" on every file in Windows if you open its properties dialog. It is growing as the hydration is progressing and its size is zero at the beginning. If one cancels the download in the Progress dialog, the "Size on disk" does not drop to zero.

Reply:
Ah nice that there's a different field in the properties dialog. Didn't spot it the last time.

So yeah I guess it needs that kill + recreate placeholder then... shame on that API, just makes no sense to not let us write to the files directly then. :-)